### ISSUE TITLE
no-map-without-usage reports false positives for class property initializers

### ISSUE THREAD

**Issue by karol-majewski** [Glavin001/tslint-clean-code#43]
Hello and thank you for your work. 

#### Example

```tsx
class Example {
  numbers: number[] = [1, 2, 3].map(x => x + 1); // Error
}
```

#### Expected result

No error — the result of mapping is assigned to a class property.

#### Actual result

```
Return value from Array.map should be assigned to a variable. Consider using Array.forEach instead. (no-map-without-usage)
```

#### Usage

- `tslint-clean-code@0.2.3`
- `tslint@5.9.1`
- `typescript@2.7.1`

**Comment by Glavin001** [comment]
Thank you for reporting this bug!

I think I know what the problem is: https://astexplorer.net/

![image](https://user-images.githubusercontent.com/1885333/36484410-913fe40a-16ef-11e8-9e48-0829d05847ea.png)

Need to add "PropertyDeclaration" to list of types for `parentUsesNode`: https://github.com/Glavin001/tslint-clean-code/blob/master/src/noMapWithoutUsageRule.ts#L87

Unit tests can be found at https://github.com/Glavin001/tslint-clean-code/blob/master/src/tests/NoMapWithoutUsageRuleTests.ts 

I would love it if you feel comfortable contributing with a Pull Request! Please let me know if you have any questions. I recommend adding a failing test case first and commit+push so Travis CI can fail. Then fix the bug by adding `PropertyDeclaration` support and watching the next commit resolve the failing case. 🎉 

Thank you in advance!

**Comment by karol-majewski** [comment]
@Glavin001 Sure, I can do that 👍

**Comment by Glavin001** [comment]
Any progress, @karol-majewski ?

**Comment by karol-majewski** [comment]
@Glavin001 I couldn't make it this week, let me try tomorrow 👍

**Comment by Glavin001** [comment]
Published to v0.2.5
